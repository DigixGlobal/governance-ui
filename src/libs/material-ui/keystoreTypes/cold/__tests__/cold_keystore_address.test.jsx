import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ColdKeystoreAddress from '../cold_keystore_address';

const data = {
  address: '0x12345',
  enabled: true,
  exists: true,
  name: 'test name',
};

describe('<ColdKeystoreAddress />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <ColdKeystoreAddress
        itemId={1}
        data={data}
        onChange={jest.fn}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
