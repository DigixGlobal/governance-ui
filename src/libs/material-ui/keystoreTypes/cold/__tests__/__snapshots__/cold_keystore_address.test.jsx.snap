// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColdKeystoreAddress /> renders correctly 1`] = `
<TableRow
  as="tr"
  cellAs="td"
  negative={false}
>
  <TableCell
    as="td"
    colSpan="2"
  >
    <AddressInput
      error={false}
      fluid={true}
      hideDropdown={true}
      name="address"
      onChange={[Function]}
      placeholder="Enter Address"
      showQrScanner={false}
      size="small"
      value="0x12345"
    />
  </TableCell>
  <TableCell
    as="td"
  >
    <Input
      disabled={false}
      error={false}
      fluid={true}
      name="name"
      onChange={[Function]}
      placeholder="Enter Name"
      size="small"
      type="text"
      value="test name"
    />
  </TableCell>
  <TableCell
    as="td"
    style={
      Object {
        "minWidth": "9em",
      }
    }
    width="1"
  >
    <Connect(NetworkTokensSelector)
      disabled={false}
      formChange={[Function]}
      formData={
        Object {
          "address": "0x12345",
          "enabled": true,
          "exists": true,
          "name": "test name",
        }
      }
      modal={true}
    />
  </TableCell>
</TableRow>
`;
